src/archive_write.cpp

```diff
@@ -73,13 +73,15 @@ void rchive_write_close(Rconnection con) {
   struct archive* out;
   struct archive_entry* entry;
   in = archive_read_disk_new();
-#ifndef __MINGW32__
+#if defined(_WIN32) || (!defined(__GNUC__) && !defined(__clang__))
```

Comment:
How is `O_BINARY` related to GCC and clang?

Reply:
I just don't trust msvc/bcc so much and am not sure if they correctly add _WIN32, but sure let's trust.

I just mean GCC/clang ensure to add _WIN32.

Reply:
@gaborcsardi FYI I dropped MSVC/clang support.